src/bin/pgcopydb/ld_test_decoding.c

```diff
@@ -319,25 +370,19 @@ parseTestDecodingMessageHeader(TestDecodingHeader *header, const char *message)
 	 *      idp   dot             sep acp   end
 	 */
 	char *idp = (char *) message + strlen("table ");
-	char *dot = strchr(idp, '.');
-	char *sep = strchr(idp, ':');
+
+	int schemaEnd = findIdentifierEndPos(idp, '.');
+	char *dot = idp + schemaEnd;
+
+	int tableEnd = findIdentifierEndPos(dot, ':');
+	char *sep = dot + tableEnd;
+
```

Comment:
What about the idea of passing a struct to fill-in and calling the function only once?

Reply:
The idea is to support identifier parsing for the following 3 scenarios with different end separator.

1) schema name parsing with separator  '.'
2) table name parsing with separator ':'
3) column name parsing with separator '['

IMHO, a single function with separator as a argument makes more sense here.

Reply:
Ah yes, you're right. Thanks!